setup.sh

```diff
@@ -142,71 +165,53 @@ function update() {
     function update_project() {
         local project=$1
         local gitproject="git://github.com/dlang/$project.git"
-        if ! ( cd "$wd/$project" && \
-            git checkout master && \
-            git pull --ff-only $gitproject master && \
-            git pull $gitproject master --tags && \
-            git fetch $gitproject && \
-            git fetch --tags $gitproject) 2>$tempdir/$project.log
+        local git="git -C $wd/$project"
+        if ! ( \
+            $git checkout master && \
+            $git pull --ff-only --tags "$gitproject" master ) 2> "$tempdir/$project.log"
```

Comment:
There was no reason to do 4x the work. `git` is smart enough to do everything in one step :)